### ISSUE TITLE
[S] Update UI for Review Data Page in A&R to match designs

### ISSUE THREAD

**Issue by kgorana** [bcgov/bcparks-ar-admin#259]
**Description**

As a BC Parks Rec Services staff I want to easily know when a recordreported in A&R is showing a large difference or is missing from previously reported values.

This ticket is to update the data review page from the first iteration to match designs.

**Acceptance Criteria:**

* A&R admin user can view and search and filter a record with a variance on a review data page in A&R 

  * filters include:

    * monthly date (no range yet) * required

Use query param date=

park * required

done. Use parkName=

single park selector for now, potential to do multi-select park in future

status of variance (unresolved, resolved, all) - not required

TBI.  Needs a filter on the variance GET endpoint.  Implement status= true/false.  Do not pass a status filter if you want all.

activity - not required

done. Use query param activity=

bundle removed based on conversation with Sally, this will diverge from designs. Reason for this is that bundle is subject to change over time upon contract renewals for POs and Park is more stable.

TBI - needs bundle attached to a variance record every time a variance is created.

sub-area - not required

done. Use subAreaId

Clear filters button exists to clear filters

View/Edit button reads “View/Edit” takes user to record on enter data form

Needs Review/Resolved button is working as expected

sysadmin can resolve/unresolved

non-sysadmin can only view

TBI.  Needs role specification. Current state is sysadmins only.

toggles like checkin status on DUP - on/off toggle

API Side implemented - PUT method body field to pass along is resolved attribute.

List is infinite scroll

Load More button at bottom of list

Review Data Page is open to all users but data  visible respects users roles

Matches designs except the items listed below which are out of scope

Notes

Out of scope:

flagging month within date picker that a variance exists (on Design 4.0 page 2, item 1)

show a default list of all variances (on Design 4.0 page 1, item 1)

export data: BRS-1198: [L] Export Report of Variances
DONE
 

Attachments (1)

Screenshot 2023-08-03 at 4.00.01PM.png
03 Aug 2023, 04:00 PM
Linked issues

blocks
Issue type: Task
[BRS-1241](https://bcparksdigital.atlassian.net/browse/BRS-1241)
Priority: Medium

TO DO

is blocked by
Issue type: Story
[BRS-898](https://bcparksdigital.atlassian.net/browse/BRS-898)
Priority: Medium
Cameron Pettit

DONE

Issue type: Buge
[BRS-1245](https://bcparksdigital.atlassian.net/browse/BRS-1245)
Priority: Medium
Jessica Wade

TO DO

Issue type: Buge
[BRS-1244](https://bcparksdigital.atlassian.net/browse/BRS-1244)
Priority: Medium
Cameron Pettit

DONE

relates to
Issue type: Design
[BRS-1106](https://bcparksdigital.atlassian.net/browse/BRS-1106)
Priority: Medium
Meredith Olsen-Maier

DONE

Issue type: Task
[BRS-1191](https://bcparksdigital.atlassian.net/browse/BRS-1191)
Priority: Medium
Mark Lisé

DONE

Issue type: Task
[BRS-1192](https://bcparksdigital.atlassian.net/browse/BRS-1192)
Priority: Medium
Mark Lisé

DONE

Issue type: Task
[BRS-1193](https://bcparksdigital.atlassian.net/browse/BRS-1193)
Priority: Medium
Daniel Truong

DONE

Issue type: Buge
[BRS-1243](https://bcparksdigital.atlassian.net/browse/BRS-1243)
Priority: Medium
Daniel Truong

DONE

Issue type: Buge
[BRS-1249](https://bcparksdigital.atlassian.net/browse/BRS-1249)
Priority: Medium
Cameron Pettit

DONE

Issue type: Task
[BRS-1250](https://bcparksdigital.atlassian.net/browse/BRS-1250)
Priority: Medium
Jessica Wade

TO DO

Designs

**Comment by kgorana** [comment]
Jessica Wade 
August 4, 2023 at 11:37 AM

UI is looking good and functioning as expected for happy path. Moving to done with awareness that highest priority bug remaining relevant to this one is BRS-1244: Adding variance notes removes offending variance fields from variance object
DONE
 .


Edit

Delete


Manuji Samara 
August 3, 2023 at 4:01 PM

Tested on TEST: Done

Verified that this user has been implemented as per the requirement and separate bugs have been created for the issues. 

Note: pending testing the “Load more” button. 


Edit

Delete


Manuji Samara 
August 3, 2023 at 4:00 PM

Discussion on Rocketchat for reference:



Edit

Delete


Manuji Samara 
August 3, 2023 at 3:44 PM

@Cameron Pettit after how many records do we display the “Load more”. button? 


Edit

Delete


Cameron Pettit 
July 26, 2023 at 1:26 PM

@Manuji Samara yes sorry


Edit

Delete


Manuji Samara 
July 26, 2023 at 12:48 PM

@Cameron Pettit is this ticket ready for QA? 


Edit

Delete


Jessica Wade 
July 19, 2023 at 2:46 PM

Discussion at planning about this ticket is to not enable date range filtering at this time, only select by month and park to simplify the search functionality as a next iteration. BRS-1241: Add Date Range search/filter to review data page 
TO DO
 would add date range.


Edit

Delete


Manuji Samara 
July 14, 2023 at 1:13 PM

Noted with thanks @Jessica Wade 


Edit

Delete


Jessica Wade 
July 14, 2023 at 12:59 PM

updated based on convo with Sally, Mark and Cam. Slight divergence from designs on the bundle filter. 
Main filter will be Monthly Date Range and Park (both required)
Then, sub-area, activity, and variance status are optional additional filters available. 

@Manuji Samara fyi the divergence on designs and clarity for testing :slight_smile: 


Edit

Delete

**Comment by kgorana** [comment]
@marklise or @JLWade can you please help me delete this issue? It was marked as 'done' which I noticed later. If we don't delete it, not a big deal, but it requires editing to make it consistent and link other issues and everything - so this (probably) might confuse users, but we can also leave it as it is - whatever you suggest.

**Comment by manuji** [comment]
@kgorana , I can close this as a duplicate.

**Comment by manuji** [comment]
Closed the ticket as per the request by @kgorana

**Comment by manuji** [comment]
Closed the ticket as per the request by @kgorana
